services/rfq/relayer/service/handlers.go

```diff
@@ -130,6 +121,39 @@ func (r *Relayer) handleBridgeRequestedLog(parentCtx context.Context, req *fastb
 	return nil
 }
 
+func (r *Relayer) getBridgeTxs(ctx context.Context, req *fastbridgev2.FastBridgeV2BridgeRequested, originClient client.EVM) (txV1 fastbridgev2.IFastBridgeBridgeTransaction, txV2 fastbridgev2.IFastBridgeV2BridgeTransactionV2, err error) {
+	fastBridge, err := fastbridgev2.NewFastBridgeV2Ref(req.Raw.Address, originClient)
+	if err != nil {
+		return txV1, txV2, fmt.Errorf("could not get correct fast bridge: %w", err)
+	}
+
+	calls := []func(ctx context.Context) error{
+		func(ctx context.Context) error {
+			txV1, err = fastBridge.GetBridgeTransaction(&bind.CallOpts{Context: ctx}, req.Request)
+			if err != nil {
+				return fmt.Errorf("could not get bridge transaction: %w", err)
+			}
+			return nil
+		},
+		func(ctx context.Context) error {
+			txV2, err = fastBridge.GetBridgeTransactionV2(&bind.CallOpts{Context: ctx}, req.Request)
```

Comment:
Don't think this affects the Relayer, but could be relevant for the Guard.

`GetBridgeTransaction` supports both V1 and V2 payloads, and decodes into the V1 struct accornigly.
`GetBridgeTransactionV2` on the other hand only supports the V2 payloads, so feeding V1 paylaod will result in a revert.